language-plutus-core/src/Language/PlutusCore/Name.hs

```diff
@@ -29,7 +33,10 @@ data Name a = Name { nameAttribute :: a
 
 newtype TyName a = TyName { unTyName :: Name a }
     deriving Show
-    deriving newtype (Eq, Functor, NFData, Pretty)
+    deriving newtype (Eq, Functor, NFData, Pretty, Debug)
+
+class Debug a where
```

Comment:
It's fine for now, but I don't think we want to use an extra type class for the debugging output in the long run. Instead, we'll have to maintain some sort of configuration environment at some point. I am saying this, because there'll likely be more options for different version of what we want to dump in the future. (GHC, for example, has quite few different switches that you want to be able to toggle individually.)

Reply:
Should I avoid the `Pretty` typeclass as well? I could make it return a result in the reader monad, where that contained configuration info.